app/javascript/App/bundles/common/components/FlashMessageListItem.tsx

```diff
@@ -0,0 +1,158 @@
+import * as React from 'react'
+import { withStyles } from '@material-ui/core/styles'
+import green from '@material-ui/core/colors/green'
+import amber from '@material-ui/core/colors/amber'
+import Slide from '@material-ui/core/Slide'
+import Snackbar, { SnackbarProps } from '@material-ui/core/Snackbar'
+import SnackbarContent from '@material-ui/core/SnackbarContent'
+import CloseIcon from '@material-ui/icons/Close'
+import classnames from 'classnames'
+import { Icon, IconButton } from '@material-ui/core'
+import { TransitionProps } from '@material-ui/core/transitions/transition'
+import { FlashMessage } from '~/bundles/common/types'
+
+const TRANSITION_DELAY = 150
+const TRANSITION_DOWN_DURATION = 200
+
+/**
+ * @param {object} anchorOrigin - how snack is postioned. e.g.
+ * { vertical: 'bottom', horizontal: 'left' }
+ * @param {number} level - Level on which snakcbar should be displayed
+ * (when snackbars are stacked on top of eachother)
+ */
+const getTransitionStyles = (level, anchorOrigin) =>
+  Object.assign(
+    {
+      [anchorOrigin.vertical]: level * 48 + level * 16 + 20,
+    },
+    {
+      WebKitTransition: `all ${TRANSITION_DOWN_DURATION}ms`,
+      MozTransition: `all ${TRANSITION_DOWN_DURATION}ms`,
+      msTransition: `all ${TRANSITION_DOWN_DURATION}ms`,
+      OTransition: `all ${TRANSITION_DOWN_DURATION}ms`,
+      transition: `all ${TRANSITION_DOWN_DURATION}ms`,
+      transitionDelay: `${TRANSITION_DELAY}ms`,
+    },
+  )
+
+const DIRECTION = {
```

Comment:
@jessehon I'm sure there's a good reason that these `DIRECTION`s are reversed but not sure off the top of my head - mind adding some comments to explain?